src/api/spec/controllers/webui/groups_controller_spec.rb

```diff
@@ -64,4 +65,79 @@
       expect(response.body).to eq([].to_json)
     end
   end
+
+  describe 'GET edit' do
+    let(:users_of_group) { create_list(:user, 3) }
+
+    before do
+      group.users << users_of_group
+
+      login(user)
+    end
+
+    context 'as a normal user' do
+      it 'does not allow to see the edit form used for updating a group' do
+        get :edit, params: { title: group.title }
+
+        expect(flash[:error]).to eq("Sorry, you are not authorized to update this Group.")
+      end
+    end
+
+    context 'as a group maintainer' do
+      before do
+        create(:group_maintainer, user: user, group: group)
+      end
+
+      it 'shows edit form and populates it with data' do
+        get :edit, params: { title: group.title }
+
+        expect(response).to have_http_status(:success)
+        assigned_members = assigns(:members).map(&:values).flatten
+        expect(assigned_members).to match_array(users_of_group.map(&:login))
+      end
+    end
+  end
+
+  describe 'POST create' do
+    let(:users_to_add) { create_list(:user, 3) }
+
+    before do
+      group.users << create(:user, login: 'existing_group_user')
+
+      login(user)
+    end
+
+    context 'as a normal user' do
+      it 'does not allow to create a group' do
+        post :create, params: { group: { title: group.title, members: users_to_add.map(&:login) }}
+
+        expect(flash[:error]).to eq("Sorry, you are not authorized to create this Class.")
+      end
+    end
+
+    context 'as an admin' do
+      before do
+        login(create(:admin_user))
+      end
+
+      context 'creating a new group' do
+        it 'creates a group with members' do
+          post :create, params: { group: { title: 'my_group', members: users_to_add.map(&:login).join(',')}}
```

Comment:
why here https://github.com/openSUSE/open-build-service/pull/2823/files#diff-587a38e2b32b8b815600f0c307b62ebeR112 you don't need the join and here you need it if it is the same method? 😕